hibernate-core/src/main/java/org/hibernate/boot/model/internal/AnnotationBinder.java

```diff
@@ -660,11 +661,14 @@ private static void bindFilterDef(FilterDef filterDef, MetadataBuildingContext c
 			throw new AnnotationException( "Multiple '@FilterDef' annotations define a filter named '" + name + "'" );
 		}
 		final Map<String, JdbcMapping> explicitParamJaMappings;
+		final Map<String, Class<? extends Supplier>> parameterResolverMap;
```

Comment:
I think we should resolve the managed beans already at this point i.e.

```suggestion
		final Map<String, ManagedBean<Supplier>> parameterResolverMap;
```

Reply:
I was troubled by this, honestly. The first thought was resolving the beans in the AnnotationBinder. I'll do the changes

Reply:
Just to be sure: if we use ManagedBeans, that will be the only way to use the resolvers: the bean context must be initialized to make the resolvers work. 
The alternative could be sticking to a Map of Suppliers. 
Which way should we go?

Reply:
>the bean context must be initialized to make the resolvers work.

Not sure what you mean by that exactly. You will just have to call `serviceRegistry.requireService( ManagedBeanRegistry.class ).getBean( supplierClass, context.getBootstrapContext().getCustomTypeProducer() )` to acquire the `ManagedBean`.

Reply:
Yep, and that's pretty clear. 
As for now, the resolver can be either a ManagedBean or a freshly created object. Tests in FilterParameterTests could clarify this further. I'm ok with sticking to ManagedBeans only, as I'm ok with maintaining the actual options. What I'm asking here is a choice on your side.

Reply:
I'm still unsure what you mean. `ManagedBean` is just an abstraction in Hibernate to abstract over how an instance of a class is acquired. If the bean container does not provide an instance, an instance for the class will be created automatically.

Reply:
What happens if the `ManagedBeanRegistry` is null, because it has not been initialized?

Reply:
That should not be possible i.e. `ManagedBeanRegistry` should always be non-null at this point.

Reply:
Great. Then I'll update the tests too and commit the changes